--- showHarvesterStatus_org	2014-10-09 08:12:08.111632906 +0200
+++ showHarvesterStatus	2014-10-09 13:34:35.563037344 +0200
@@ -123,10 +123,11 @@
     #
 %>
 <tr class="eventinfo">
-<td><a name="<%= repositoryId %>" href="#top_<%= repositoryId %>"><%= repositoryId %></a></td>
+<td><a name="<%= repositoryId %>" href="#top_<%= repositoryId %>"><%= repositoryId %></a></td><td>&nbsp;</td>
 </tr>
 <tr>
-<td>Nr. of validation errors: <strong><%= invalidnr %></strong></td>
+<td>Nr. of validation errors: </td>
+<td><strong><%= invalidnr %></strong></td>
 </tr>
 <%
     if hasattr(status.recentinvalids, 'invalidId'):
@@ -139,6 +140,9 @@
           <td class="link">
             <a href="/page/invalidRecord/?<%= urlencode(newArgs)%>"><%= recordId%></a>
           </td>
+          <td>          
+          <% include(req.get_options()['internalServer'] + '/invalidRecord_gmh?' + urlencode(newArgs)) %>
+          </td>
         </tr>
 <%
 #
